Common/GPU/ShaderWriter.h

```diff
@@ -75,10 +75,12 @@ class ShaderWriter {
 	void HighPrecisionFloat();
 	void LowPrecisionFloat();
 
+	// NOTE: samplers must live for the rest of ShaderWriter's lifetime. No way to express that in C++ though :(
 	void DeclareSamplers(Slice<SamplerDef> samplers);
```

Comment:
While I agree it'd be nice if it could track borrowing moreso, it wouldn't really be that expensive to make a copy here and that'd make it safe.

-[Unknown]

Reply:
Fair enough, heh.